Library/ViewModels/PostCampaignPledgeRewardsSummaryViewModel.swift

```diff
@@ -0,0 +1,223 @@
+    case .header: return true
+    case .reward: return false
+    }
+  }
+
+  public var isReward: Bool {
+    switch self {
+    case .header: return false
+    case .reward: return true
+    }
+  }
+}
+
+public struct PostCampaignRewardsSummaryViewData {
+  public let rewards: [Reward]
+  public let selectedQuantities: SelectedRewardQuantities
+  public let projectCountry: Project.Country
+  public let omitCurrencyCode: Bool
+  public let shipping: PledgeShippingSummaryViewData
+}
+
+public protocol PostCampaignPledgeRewardsSummaryViewModelInputs {
+  func configureWith(rewardsData: PostCampaignRewardsSummaryViewData, bonusAmount: Double?,
+                     pledgeData: PledgeSummaryViewData)
+  func viewDidLoad()
+}
+
+public protocol PostCampaignPledgeRewardsSummaryViewModelOutputs {
+  var configurePledgeTotalViewWithData: Signal<PledgeSummaryViewData, Never> { get }
+  var loadRewardsIntoDataSource: Signal<[PostCampaignRewardsSummaryItem], Never> { get }
+}
+
+public protocol PostCampaignPledgeRewardsSummaryViewModelType {
+  var inputs: PostCampaignPledgeRewardsSummaryViewModelInputs { get }
+  var outputs: PostCampaignPledgeRewardsSummaryViewModelOutputs { get }
+}
+
+public final class PostCampaignPledgeRewardsSummaryViewModel: PostCampaignPledgeRewardsSummaryViewModelType,
+  PostCampaignPledgeRewardsSummaryViewModelInputs, PostCampaignPledgeRewardsSummaryViewModelOutputs {
+  public init() {
+    let data = Signal.combineLatest(
+      self.configureWithDataProperty.signal.skipNil(),
+      self.viewDidLoadProperty.signal
+    )
+    .map(first)
+
+    let rewardData = data.map { $0.0 }
+    let rewards = data.map(\.0.rewards)
+    let selectedQuantities = data.map(\.0.selectedQuantities)
+    let bonusAmount = data.map(\.1)
+
+    let latestRewardDeliveryDate = rewards.map { rewards in
+      rewards
+        .compactMap { $0.estimatedDeliveryOn }
+        .reduce(0) { accum, value in max(accum, value) }
+    }
+
+    let estimatedDeliveryString = latestRewardDeliveryDate.map { date -> String? in
+      guard date > 0 else { return nil }
+
+      let dateString = Format.date(
+        secondsInUTC: date,
+        template: DateFormatter.monthYear,
+        timeZone: UTCTimeZone
+      )
+
+      return Strings.backing_info_estimated_delivery_date(delivery_date: dateString)
+    }
+    .skipNil()
+
+    let total: Signal<Double, Never> = Signal.combineLatest(
+      rewards,
+      selectedQuantities
+    )
+    .map { rewards, selectedQuantities in
+      rewards.reduce(0.0) { total, reward in
+        let totalForReward = reward.minimum
+          .multiplyingCurrency(Double(selectedQuantities[reward.id] ?? 0))
+
+        return total.addingCurrency(totalForReward)
+      }
+    }
+
+    self.loadRewardsIntoDataSource = Signal.zip(
+      rewardData,
+      selectedQuantities,
+      estimatedDeliveryString,
+      bonusAmount,
+      total
+    )
+    .map(items)
+
+    self.configurePledgeTotalViewWithData = data.map { $0.2 }
```

Comment:
Nit: This view is complicated enough I've already forgotten what `$0.2` would be. Can this be clarified, either by naming the values in the tuple and/or by naming the variable here instead of using `$0`?